plugins/flytekit-xgboost/tests/test_task.py

```diff
@@ -0,0 +1,176 @@
+import os
+from dataclasses import asdict, dataclass
+from typing import Dict, List, NamedTuple, Tuple
+
+import numpy as np
+import pandas as pd
+import xgboost
+from dataclasses_json import dataclass_json
+from flytekitplugins.xgboost import TrainParameters, XGBoostTask
+from sklearn import model_selection
+
+import flytekit
+from flytekit import kwtypes, task, workflow
+from flytekit.types.file import CSVFile, FlyteFile, JoblibSerializedFile
+
+
+@dataclass_json
+@dataclass
+class Hyperparameters:
+    max_depth: int = 2
+    eta: int = 1
+    objective: str = "binary:logistic"
+    verbosity: int = 2
+
+
+@dataclass_json
+@dataclass
+class CSVHyperparameters:
+    objective: str = "reg:linear"
+    eta: float = 0.2
+    gamma: int = 4
+    max_depth: int = 5
+    subsample: float = 0.7
+    silent: int = 0
+    min_child_weight: int = 6
+
+
+def test_simple_model():
+    task_obj = XGBoostTask(
+        name="test1",
+        hyperparameters=asdict(Hyperparameters()),
+        task_config=TrainParameters(num_boost_round=2),
+        inputs=kwtypes(train=FlyteFile, test=FlyteFile),
+    )
+
+    @workflow
+    def train_test_wf(
+        train: FlyteFile = "https://raw.githubusercontent.com/dmlc/xgboost/master/demo/data/agaricus.txt.train",
+        test: FlyteFile = "https://raw.githubusercontent.com/dmlc/xgboost/master/demo/data/agaricus.txt.test",
+    ) -> List[float]:
+        _, predictions, _ = task_obj(train=train, test=test)
+        return predictions
+
+    assert task_obj.python_interface.inputs == {"train": FlyteFile, "test": FlyteFile}
+    train_test_wf()
+
+
+def test_csv_data():
+    """
+    Note: For CSV training, the algorithm assumes that the CSV does not have a header record.
+    To set the target variable, set the label_column parameter, which by default is 0.
+    """
+    task_obj = XGBoostTask(
+        name="test2",
+        hyperparameters=asdict(CSVHyperparameters()),
+        inputs=kwtypes(train_data=CSVFile, test_data=CSVFile),
+        label_column=0,
+    )
+
+    @task
+    def partition_data(dataset: str) -> Tuple[CSVFile, CSVFile]:
```

Comment:
why CSV, why not pandas dataframes or Better FlyteSchema

Reply:
We can also use FlyteSchema; that way, we could support FlyteFile and FlyteSchema as valid data types. But why do we want to convert CSV to Schema internally if we have CSVFile as the supported Flyte type? XGBoost supports CSV as well.